material-db-tools/material_db_tools.py

```diff
@@ -53,3 +56,39 @@ def mix_by_volume(material_library, vol_fracs, citation, density_factor=1):
     )
     return mat
 
+
+def enrich(
+    material_composition: Dict[Union[int, str], float],
+    element_to_enrich: Union[int, str],
+    isotope_enrichments: Dict[Union[int, str], float],
+) -> Dict[int, float]:
+    """
+    Enrich a specific element in a material composition by replacing it with its isotopes.
+
+    Args:
+        material_composition (Dict[Union[int, str], float]): The original material composition.
+            Keys can be nuclide IDs (int) or element symbols (str), and values are their fractions (float).
+        element_to_enrich (Union[int, str]): The ID or symbol of the element to be enriched.
+        isotope_enrichments (Dict[Union[int, str], float]): The isotopic composition for element_to_enrich.
+            Keys can be isotope IDs (int) or isotope symbols (str), and values are their enrichment fractions (float).
+
+    Returns:
+        Dict[int, float]: The updated material composition with the enriched element.
+            Keys are nuclide IDs (int) and values are their fractions (float).
+    """
+    # Convert all keys to PyNE nuclide IDs
+    material_composition = {
+        nucname.id(k): Decimal(str(v)) for k, v in material_composition.items()
+    }
+    isotope_enrichments = {
+        nucname.id(k): Decimal(str(v)) for k, v in isotope_enrichments.items()
+    }
+    element_to_enrich_id = nucname.id(element_to_enrich)
+
+    if element_to_enrich_id in material_composition:
+        fract = material_composition.pop(element_to_enrich_id)
+        for nuclide, enrich_frac in isotope_enrichments.items():
+            material_composition[nuclide] = enrich_frac * fract
```

Comment:
For the example you provided, a user could specify their original material which is 10% Li by mass, then convert the material to atomic fraction. Then if they enriched either material using make_mat_from_atom that would allow them to easily maintain their atomic ratio while enriching the element. I ran a test based on your example using Li2O3Ti with 10% mass lithium and enriching it to 0 and 100%. The elemental mass fraction of lithium stayed at 10%. The atomic mass fraction of lithium changed by 3.5% from the 0% to 100% lithium 6 enrichment case, or roughly +-5% relative to the overall atomic mass.  We could tell them that the enrichment function does not currently take into consideration changes in density that result from enrichment and that the density provided will be the final material density. The user-specified fractions for either case are what the material's final composition is.